src/integration/java/io/pinecone/helpers/TestIndexResourcesManager.java

```diff
@@ -108,14 +109,15 @@ private String createOrGetPodIndex() throws InterruptedException, PineconeExcept
 
         String indexName = RandomStringBuilder.build("pod-index", 8);
 
-        podIndexModel = pineconeClient.createPodsIndex(indexName, dimension, environment, "p1.x1");
+        podIndexModel = pineconeClient.createPodsIndex(indexName, dimension, environment, "p1.x1", metric);
         waitUntilIndexIsReady(pineconeClient, indexName);
 
-        // Explicitly wait after ready to avoid the "no healthy upstream" issue
+        // Additional sleep after index marked as ready to avoid "no healthy upstream" error
         Thread.sleep(30000);
 
         // Seed data
-        seedIndex(podIndexVectorIds, indexName);
+        Index indexClient = pineconeClient.getIndexConnection(indexName);
```

Comment:
W/[my latest PR](https://github.com/pinecone-io/pinecone-java-client/pull/115), I've introduced `getServerlessIndexConnection` and `getServerlessAsyncIndexConnection`. I think you should add a `getPodsIndexConnection` to make things more symmetrical? But not mandatory obvi.